ui/article-renderer/src/main/java/org/cru/godtools/article/ui/ArticlesActivity.kt

```diff
@@ -0,0 +1,104 @@
+package org.cru.godtools.article.ui
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import androidx.annotation.MainThread
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.commit
+import dagger.hilt.android.AndroidEntryPoint
+import java.util.Locale
+import org.ccci.gto.android.common.androidx.lifecycle.notNull
+import org.ccci.gto.android.common.androidx.lifecycle.observeOnce
+import org.cru.godtools.article.R
+import org.cru.godtools.article.aem.model.Article
+import org.cru.godtools.article.aem.ui.startAemArticleActivity
+import org.cru.godtools.article.ui.articles.ArticlesFragment
+import org.cru.godtools.article.ui.categories.CategoriesFragment
+import org.cru.godtools.article.ui.categories.CategorySelectedListener
+import org.cru.godtools.base.tool.activity.BaseArticleActivity
+import org.cru.godtools.base.tool.activity.BaseSingleToolActivity
+import org.cru.godtools.base.tool.databinding.ToolGenericFragmentActivityBinding
+import org.cru.godtools.xml.model.Category
+
+fun Context.createArticlesIntent(toolCode: String, language: Locale): Intent {
+    return Intent(this, ArticlesActivity::class.java)
+        .putExtras(BaseSingleToolActivity.buildExtras(this, toolCode, language))
+}
+
+fun Activity.startArticlesActivity(toolCode: String, language: Locale) {
+    startActivity(createArticlesIntent(toolCode, language))
+}
+
+@AndroidEntryPoint
+class ArticlesActivity :
+    BaseArticleActivity<ToolGenericFragmentActivityBinding>(R.layout.tool_generic_fragment_activity),
+    CategorySelectedListener,
+    ArticlesFragment.Callbacks {
+    // region Lifecycle
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (isFinishing) return
+        if (savedInstanceState == null) trackToolOpen(tool)
+        setupFragments()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem) = when {
+        item.itemId == android.R.id.home && supportFragmentManager.backStackEntryCount > 0 -> {
+            supportFragmentManager.popBackStack()
+            true
+        }
+        else -> super.onOptionsItemSelected(item)
+    }
+
+    override fun onCategorySelected(category: Category?) = showFragment(articlesFragment(category), true)
```

Comment:
I saw there was logic in the ```CategoriesActivity``` to only load the fragment if needed, but I guess it still had to run logic to check if it was needed. This solution is better because it's only loaded once the category is selected and the logic to check is no longer needed right?

Reply:
the `loadFragmentIfNecessary()` logic would create the fragment the first time the activity is created, but not recreate it if the activity was recreated.

This logic still exists [here](https://github.com/CruGlobal/godtools-android/pull/1596/files#diff-bf7f611f9f31954e845ae85b090a1a5e3987f71f0f7a6ce9dab07a23792e87d4R78-R86) but now intelligently chooses what that initial fragment being loaded is